src/alchemlyb/tests/test_preprocessing.py

```diff
@@ -78,6 +78,37 @@ def test_multiindex_duplicated(self, gmx_ABFE):
                                              gmx_ABFE.sum(axis=1))
         assert len(subsample) == 501
 
+    def test_sort(self, gmx_ABFE):
```

Comment:
split in two tests, one for unsorted and KeyError, the other for sorted. More granular tests are better and failing the first stops testing the second in the current form